File: Assignment2/pegasos-multi/data/show_img.py
# The script loads the training data and shows the first image.
# This may be helpful for visualizing the data and show how to load the data using scipy

import scipy as sp
import matplotlib.pyplot as plt
train = sp.genfromtxt('mnist_train.txt', delimiter=',')
img = train[0, 1:].astype(sp.uint8).reshape(28, 28)
plt.gray()
plt.imshow(img)
plt.show()

File: Assignment2/pegasos-multi/data/show_img.m
% The script loads the training data and show an image of a random index.
% This may be helpful for visualizing the data and show how to load the data in Matlab or Octave

train = load('mnist_train.txt');
idx = randi(size(train, 1));
img = uint8(reshape(train(idx , 2:end), 28, 28)');
imshow(img)
